src/components/microlearning/VideoCourseReview.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Icon, OverlayTrigger, Tooltip } from '@openedx/paragon';
+import { StarFilled, StarOutline } from '@openedx/paragon/icons';
+import PropTypes from 'prop-types';
+import { FormattedMessage, useIntl } from '@edx/frontend-platform/i18n';
+import { useVideoCourseReviews } from '../app/data';
+import { fixDecimalNumber } from '../course/data';
+
+const VideoCourseReview = ({ courseKey }) => {
+  const { data: courseReviews } = useVideoCourseReviews(courseKey);
+  const intl = useIntl();
+  if (!courseReviews) {
+    return null;
+  }
+  const rating = Math.round(fixDecimalNumber(courseReviews.avgCourseRating)); // Round to the nearest whole number
```

Comment:
The statement is likely to throw an error if `avgCourseRating` is undefined. Additionally, passing an undefined value to `fixDecimalNumber` will result in a TypeError because the function calls .toFixed() on the result of parseFloat(), and calling .toFixed() on NaN causes a type error.



Suggestion
`const rating = Math.round(fixDecimalNumber(courseReviews?.avgCourseRating ?? 0));`

Reply:
Actually, i have replicated the existing implemented logic on a course about page. Too much unnecessary optional chaining will make our code slow. That's why I tried to use minimal optional chaining.

Reply:
@jajjibhai008 I think we should add necessary checks for potential errors wherever required and optional chaining has negligible performance impact in modern JavaScript engines, it does not make the system slow.